test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlServerTaskConsumerTests.cs

```diff
@@ -0,0 +1,373 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.TaskManagement;
+using Microsoft.Health.Fhir.SqlServer.Features.Storage;
+using Microsoft.Health.Fhir.Tests.Integration.Persistence;
+using Newtonsoft.Json;
+using NSubstitute;
+using Xunit;
+using TaskStatus = Microsoft.Health.Fhir.Core.Features.TaskManagement.TaskStatus;
+
+namespace Microsoft.Health.Fhir.Shared.Tests.Integration.Persistence
+{
+    public class SqlServerTaskConsumerTests : IClassFixture<SqlServerFhirStorageTestsFixture>
+    {
+        private SqlServerFhirStorageTestsFixture _fixture;
+
+        public SqlServerTaskConsumerTests(SqlServerFhirStorageTestsFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        [Fact]
+        public async Task GivenListOfTasksInQueue_WhenGetNextTask_AvailableTasksShouldBeReturned()
+        {
+            string queueId = Guid.NewGuid().ToString();
+            TaskHostingConfiguration config = new TaskHostingConfiguration()
+            {
+                Enabled = true,
+                QueueId = queueId,
+                TaskHeartbeatTimeoutThresholdInSeconds = 60,
+            };
+
+            IOptions<TaskHostingConfiguration> optionsExportConfig = Substitute.For<IOptions<TaskHostingConfiguration>>();
```

Comment:
>optionsExportConfig [](start = 47, length = 19)

nit: maybe rename to `taskHostingConfig` ?

Reply:
Thanks, updated.